src/lcals/FIRST_MIN-Cuda.cpp

```diff
@@ -80,21 +81,36 @@ void FIRST_MIN::runCudaVariantImpl(VariantID vid)
 
        FIRST_MIN_MINLOC_INIT;
 
+       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
+
+       MyMinLoc mymin_block[grid_size]; //per-block min value
+       for (Index_type i = 0; i < static_cast<Index_type>(grid_size); i++) {
+         mymin_block[i] = mymin;
+       }
+
        MyMinLoc* dminloc;
-       cudaErrchk( cudaMalloc( (void**)&dminloc, sizeof(MyMinLoc) ) );
-       cudaErrchk( cudaMemcpy( dminloc, &mymin, sizeof(MyMinLoc),
-                               cudaMemcpyHostToDevice ) );
+       cudaErrchk( cudaMalloc( (void**)&dminloc, 
+                               grid_size * sizeof(MyMinLoc) ) );
```

Comment:
Can we move some of this out of the inner irep loop?

Reply:
Moved some init functions outside the run rep loop. Let me know if that was what you were thinking? I'm debating about what is a fair comparisons with other variants.

Reply:
I'm working on a few changes that I think Jason had in mind. I will push in a bit. 

It looks there is a memory leak here since the device malloc of dminloc is inside the irep loop, but the free is outside of the loop. I will address in my changes.

Reply:
Its fair in my opinion to allocate outside the loop as RAJA uses a memory pool instead of allocating new memory each time.